WalletWasabi.Fluent/Extensions/CurrencyExtensions.cs

```diff
@@ -25,10 +25,22 @@ public static Money CalculateDestinationAmount(this BuildTransactionResult resul
 		}
 		else
 		{
-			return result.InnerWalletOutputs
+			Money destinationAmount = result.InnerWalletOutputs
 				.Where(x => !x.HdPubKey.IsInternal)
 				.Select(x => x.Amount)
 				.Sum();
+
+			// There was no external address in the TX. Internal address re-use case.
+			if (destinationAmount == Money.Zero)
+			{
+				// We sum the Used key's amount, because the clean internal key in the TX is the change.
+				destinationAmount = result.InnerWalletOutputs
+				.Where(x => x.HdPubKey.KeyState is Blockchain.Keys.KeyState.Used)
+				.Select(x => x.Amount)
+				.Sum();
```

Comment:
Do we need to call `Sum()` here? 
If we want to get the amount which was sent to the used internal key (the destination address), there should be one of that. So instead of `Where`, we could use `Single` and return the amount.

Same logic could be used above.